docs/ref/forms/models.txt

```diff
@@ -1,15 +1,258 @@
-:doc:`/topics/forms/modelforms` topic guide.
+===========
+ModelForms
+===========
 
 .. module:: django.forms.models
-   :synopsis: Django's functions for building model forms and formsets.
+    :synopsis: ModelForm API reference for inner ``Meta`` class and factory
+        functions
+
+.. class:: ModelForm
+
+.. _ref-modelform:
+
+.. currentmodule:: django.forms
+
+:class:`~django.forms.ModelForm` API reference. For introductory material about
+ModelForms, see the :doc:`/topics/forms/modelforms` topic guide.
+
+.. _ref-modelform-meta-options:
+
+ModelForm ``Meta`` Options
+==========================
+
+The :class:`~django.forms.ModelForm` helper class can be used to build forms
+that closely reflect specific Django models. The exact structure of the form
+produced can be controlled by setting metadata options as attributes of an inner
+``Meta`` class.
+
+::
+
+    # Example using some sort of Book model
+
+    from django.forms import ModelForm
+    from myLibraryApp.models import Book
+
+
+    class BookForm(ModelForm):
+        class Meta:
+            model = Book
+            fields = ["title", "author"]
+            help_texts = {
+                "title": "The title of the book",
+                "author": "The author of the book",
+            }
+            # ... other attributes
+
+.. admonition:: Invalid Meta class attributes
+
+    While any number of attributes can be included in a ``Meta``
+    class, **only attributes that match those described below will be used**.
+    When troubleshooting or encountering unexpected behavior, be sure to check
+    for typos or misnamed attributes because these will **not**, in and of
+    themselves, raise an error (though they might cause another error to be
+    raised if they affect a required attribute).
+
+Other than :attr:`~ModelFormOptions.model` and at least one of
+:attr:`~ModelFormOptions.fields` and :attr:`~ModelFormOptions.exclude`, ``Meta``
+attributes are optional. It is **not** necessary for every option to be set for
+each ``Meta`` class, nor does every included field need to be included in every
+option used. If an option isn't set, or if a model's field isn't used with a
+particular option, Django will fall back on whatever has already been defined
+for a field in their model or whatever defaults might exist for that field's
+type.
+
+Optional ``Meta`` attributes, other than
+:attr:`~ModelFormOptions.localized_fields` and
+:attr:`~ModelFormOptions.formfield_callback`, expect a dictionary that maps a
+model field name string to some other object or string. Including invalid or
+excluded field name strings in these dictionaries is possible but should have no
+effect since fields that haven't been included would never be accessed.
+
+``error_messages``
+------------------
+
+.. attribute:: ModelFormOptions.error_messages
+
+    Expects a dictionary that maps a model field name string to a dictionary of
```

Comment:
I noticed that there are some other areas of the documentation that are similar, but use different language. Specifically in `ModelAdmin`'s reference docs (https://django--18699.org.readthedocs.build/en/18699/ref/contrib/admin/index.html#django.contrib.admin.ModelAdmin.prepopulated_fields). It looks like most places uses something like "Set <field_name> to ..."

Following that same pattern may make translations easier down the line because it'll be similar phrasing. It should require less consideration of what the English phrase means since there are other cases of it.

This is a minor thing I noticed, not sure if it needs to change though.

Reply:
```suggestion
    A dictionary that maps a model field name to a dictionary of error
```
True, my personal preference is it just stating what it is :grin: